openmls/src/group/managed_group/mod.rs

```diff
@@ -1086,50 +1105,82 @@ impl ManagedGroup {
     }
 }
 
-/// Unified message type
+/// Unified message type for input to the managed API
+#[derive(Debug, Clone)]
+pub enum MlsMessageIn<'a> {
+    /// An OpenMLS `MlsPlaintext`.
+    Plaintext(VerifiableMlsPlaintext<'a>),
+
+    /// An OpenMLS `MlsCiphertext`.
+    Ciphertext(MlsCiphertext),
+}
+
+#[cfg(any(feature = "test-utils", test))]
+impl<'a> MlsMessageIn<'a> {
+    pub fn group_id(&self) -> &[u8] {
+        match self {
+            MlsMessageIn::Ciphertext(m) => m.group_id().as_slice(),
+            MlsMessageIn::Plaintext(m) => m.group_id().as_slice(),
+        }
+    }
+}
+
+/// Unified message type for output by the managed API
 #[derive(PartialEq, Debug, Clone)]
-pub enum MlsMessage {
+pub enum MlsMessageOut {
     /// An OpenMLS `MlsPlaintext`.
     Plaintext(MlsPlaintext),
 
     /// An OpenMLS `MlsCiphertext`.
     Ciphertext(MlsCiphertext),
 }
 
-impl From<MlsPlaintext> for MlsMessage {
+impl From<MlsPlaintext> for MlsMessageOut {
     fn from(mls_plaintext: MlsPlaintext) -> Self {
-        MlsMessage::Plaintext(mls_plaintext)
+        MlsMessageOut::Plaintext(mls_plaintext)
     }
 }
 
-impl From<MlsCiphertext> for MlsMessage {
+impl From<MlsCiphertext> for MlsMessageOut {
     fn from(mls_ciphertext: MlsCiphertext) -> Self {
-        MlsMessage::Ciphertext(mls_ciphertext)
+        MlsMessageOut::Ciphertext(mls_ciphertext)
     }
 }
 
-impl MlsMessage {
+impl MlsMessageOut {
     /// Get the group ID as plain byte vector.
     pub fn group_id(&self) -> &[u8] {
         match self {
-            MlsMessage::Ciphertext(m) => m.group_id().as_slice(),
-            MlsMessage::Plaintext(m) => m.group_id().as_slice(),
+            MlsMessageOut::Ciphertext(m) => m.group_id().as_slice(),
+            MlsMessageOut::Plaintext(m) => m.group_id().as_slice(),
         }
     }
 
     /// Get the epoch as plain u64.
     pub fn epoch(&self) -> u64 {
         match self {
-            MlsMessage::Ciphertext(m) => m.epoch.0,
-            MlsMessage::Plaintext(m) => m.epoch().0,
+            MlsMessageOut::Ciphertext(m) => m.epoch.0,
+            MlsMessageOut::Plaintext(m) => m.epoch().0,
         }
     }
 
     /// Returns `true` if this is a handshake message and `false` otherwise.
     pub fn is_handshake_message(&self) -> bool {
         match self {
-            MlsMessage::Ciphertext(m) => m.is_handshake_message(),
-            MlsMessage::Plaintext(m) => m.is_handshake_message(),
+            MlsMessageOut::Ciphertext(m) => m.is_handshake_message(),
+            MlsMessageOut::Plaintext(m) => m.is_handshake_message(),
```

Comment:
So this is never called with a plaintext according to codecov? If so, we should add a test case.